scripts/enforce-binary-compatibility

```diff
@@ -22,23 +19,40 @@ plugin_zip=$(echo frontendUi/build/distributions/git-machete-intellij-plugin-${p
 set -x
 
 # See https://github.com/JetBrains/intellij-plugin-verifier
-verifier_jar=verifier-all.jar
-if ! [[ -f $verifier_jar ]]; then
-  curl -fsL -o $verifier_jar https://dl.bintray.com/jetbrains/intellij-plugin-service/org/jetbrains/intellij/plugins/verifier-cli/${verifier_version}/verifier-cli-${verifier_version}-all.jar
+verifier_dir="$HOME/.cache/intellij-verifier"
+mkdir -p $verifier_dir
+
+verifier_jar_path=$verifier_dir/verifier-all.jar
+if ! [[ -f $verifier_jar_path ]]; then
+  curl -fsL -o $verifier_jar_path https://dl.bintray.com/jetbrains/intellij-plugin-service/org/jetbrains/intellij/plugins/verifier-cli/${verifier_version}/verifier-cli-${verifier_version}-all.jar
 fi
 
 for iv in ${intellij_versions[*]}; do
   year=20${iv:0:2}
   release=${iv:2:1}
-  idea_dir=idea-IC-${iv}
+  idea_dir=$verifier_dir/idea-IC-${iv}
   if ! [[ -d $idea_dir ]]; then
     # Released packages apparently aren't available under version.build file name, only under year.release name
     # EAP packages apparently aren't available under year.release file name, only under version.build name
-    {
-      curl -fsL https://download.jetbrains.com/idea/ideaIC-${year}.${release}.tar.gz \
-      || curl -fsL https://download.jetbrains.com/idea/ideaIC-${iv}.tar.gz
-    } | tar xzf -
+    curl -fsL -o $idea_dir.tar.gz https://download.jetbrains.com/idea/ideaIC-${year}.${release}.tar.gz \
+    || curl -fsL -o $idea_dir.tar.gz https://download.jetbrains.com/idea/ideaIC-${iv}.tar.gz
+
+    # This hack is necessary in case when downloaded tar.gz has different name than dir inside this tar.gz
+    # like in case of 193.6911.18
+    mkdir -p $idea_dir
+    tar xzf $idea_dir.tar.gz -C $idea_dir --strip-components 1
+    rm -f $idea_dir.tar.gz
   fi
 
-  java -jar "$verifier_jar" check-plugin -vrd "frontendUi/build/verification-$(date +%F_%T)" -runtime-dir "${idea_dir}/jbr/bin" "${plugin_zip}" "${idea_dir}"
+  OUT=$(java -jar "$verifier_jar_path" check-plugin -vrd "frontendUi/build/verification-$(date +%F_%T)" -runtime-dir "${idea_dir}/jbr/bin" "${plugin_zip}" "${idea_dir}")
+  PROBLEMS=$(echo "$OUT" | grep -c -E "^Plugin (.*) against .*: .* compatibility problems?$" || true)
+  if [[ $PROBLEMS -gt 0 ]]
```

Comment:
Nit: could also be:
```suggestion
  if echo "$OUT" | grep -qxE 'Plugin (.*) against .*: .* compatibility problems?'; then
```
but the current version is also ok

Reply:
nice!